### PR TITLE
Clean up SQL involving access

### PR DESCRIPTION
**Pull request by pnorman**
As part of my hstore work, I'm cleaning up some existing SQL to make it easier to keep a new branch in sync.

### CODE REVIEW

**Comment by pnorman** [comment]
https://github.com/gravitystorm/openstreetmap-carto/pull/1293/files?w=1 is a slightly cleaner view of the changes

File: project.yaml [hunk 1]
```diff
@@ -705,7 +706,10 @@ Layer:
                 way, prio,
                 ('highway_' || (CASE WHEN substr(highway, length(highway)-3, 4) = 'link' THEN substr(highway, 0, length(highway)-4) ELSE highway END)) AS feature,
                 horse, foot, bicycle, tracktype,
-                CASE WHEN access IN ('destination') THEN 'destination'::text WHEN access IN ('no', 'private') THEN 'no'::text ELSE NULL END AS access, construction,
```
Comment (math1985) [inline project.yaml]: Did you drop the construction key deliberately?

Comment (pnorman) [inline project.yaml]: no

Comment (math1985) [inline project.yaml]: If you fix this I can merge the PR.

Comment (pnorman) [inline project.yaml]: fixed

**Pull request merged**